addons/binding/org.openhab.binding.geofence/ESH-INF/thing/device.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="geofence"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<!-- Thing Types -->
+	<thing-type id="device">
+		<label>Tracker device</label>
+		<description>Device running trackerType application</description>
+		<channels>
+			<channel id="lastReport" typeId="lastReport"/>
+			<channel id="location" typeId="location"/>
+			<channel id="batteryLevel" typeId="batteryLevel"/>
+		</channels>
+		<config-description-ref uri="thing-type:geofence:device"/>
+	</thing-type>
+	<!-- Channel Types -->
+	<channel-type id="lastReport">
+		<item-type>Date</item-type>
+		<label>Last seen</label>
+		<description>Last report timestamp</description>
+		<state pattern="%1$tF %1$tR" readOnly="true"/>
+	</channel-type>
+	<channel-type id="location">
+		<item-type>Location</item-type>
+		<label>Location</label>
+		<description>Tracker location</description>
+		<state pattern="%s" readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryLevel">
+		<item-type>Number</item-type>
+		<label>Tracker battery level [%]</label>
+		<description>Battery level in percent</description>
+		<state pattern="%d" readOnly="true"/>
+	</channel-type>
+	<channel-type id="distance">
+		<item-type>Number</item-type>
+		<label>Distance</label>
+		<description>Distance from primary location</description>
+		<state pattern="%.2f" readOnly="true"/>
+	</channel-type>
+	<channel-type id="regionPresence">
+		<item-type>Switch</item-type>
+		<label>Region switch</label>
+		<description>ON in case the device is at specific location, OFF otherwise. This channel is added to the thing at
+		runtime is the region is configured to have a switch for region presence instead of using triggers.</description>
+		<state pattern="%s" readOnly="true"/>
+	</channel-type>
+	<channel-type id="networkPresence">
+		<item-type>Switch</item-type>
+		<label>Local network</label>
+		<description>ON in case the device is on the same local network as the openHAB server, OFF otherwise. This
+		channel is added to the thing at runtime if the device IP address is static on the local network. This requires
+		DHCP configuration on the home network router as well.</description>
+		<state pattern="%s" readOnly="true"/>
+	</channel-type>
+	<channel-type id="regionEnterTrigger">
+		<kind>trigger</kind>
+		<label>Region enter trigger channel</label>
+		<description>Trigger channel for entering regions. Payload is the region name. It will be
+		created dynamically at runtime if at least on region configured for device (central or device level configuration)
+		requires a trigger to be fired when entering the region.</description>
+		<event/>
+	</channel-type>
+	<channel-type id="regionLeaveTrigger">
```

Comment:
I think it will make rules more complex as you will have to parse the payload before actions. E.g. if you want OH to notify you on speakers (of course using the new Google Cloud TTS service 😄 ) about the region your daughter arrived to you need to trim the <> prefix first and format the message based on prefix. On the other hand it is pretty simple with dedicated trigger channels.
I think this approach will increase the complexity on the end user side instead of dealing with it inside the binding BUT I can live with that.

Reply:
I think trigger channel per region would be too redundant... I implemented the <> version and will see...